packages/grafana-ui/src/components/GraphNG/utils.ts

```diff
@@ -27,8 +27,41 @@ function applySpanNullsThresholds(frames: DataFrame[]) {
   return frames;
 }
 
+// sets to undefined all same future values not preceeded by explicit null
+// in:  1,        1,undefined,        1,2,        2,null,2,3
+// out: 1,undefined,undefined,undefined,2,undefined,null,2,3
+function unsetSameFutureValues(frames: DataFrame[]) {
+  for (const frame of frames) {
+    for (const field of frame?.fields!) {
+      if (field.type !== FieldType.time) {
+        let mergeValues = field.config.custom?.mergeValues;
+
+        if (mergeValues) {
+          let values = field.values.toArray();
+          let prevVal = values[0];
+
+          for (let i = 1; i < values.length; i++) {
+            let value = values[i];
+
+            if (value == null) {
+              prevVal = null;
+            } else {
+              if (prevVal != null && value === prevVal) {
+                values[i] = undefined;
+              }
+              prevVal = value;
+            }
+          }
+          field.values = new ArrayVector(values);
+        }
+      }
+    }
+  }
+}
+
 export function preparePlotFrame(frames: DataFrame[], dimFields: XYFieldMatchers) {
   applySpanNullsThresholds(frames);
+  unsetSameFutureValues(frames);
```

Comment:
can this only apply to timeline?

Reply:
yeah, and the one above it is pretty timeseries-specific.